### ISSUE TITLE
Allow to set SourceIdentity in assume_role configuration

### ISSUE THREAD

**Issue by chgerkens** [hashicorp/terraform-provider-aws#22128]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Please provide a way to specify the __SourceIdentity__ argument as part of an assume_role configuration.

### New or Affected Resource(s)

* provider aws, assume_role block: https://registry.terraform.io/providers/hashicorp/aws/latest/docs#assume_role-configuration-block

### Potential Terraform Configuration

I suggest to add a new argument "source_identity" to the assume_role configuration block:
```hcl
provider "aws" {
  assume_role {
    role_arn     = "arn:aws:iam::ACCOUNT_ID:role/ROLE_NAME"
    session_name = "SESSION_NAME"
    external_id  = "EXTERNAL_ID"
    source_identity = "SOURCE_IDENTITY" // suggested new attribute
  }
}
```

### References

* https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRole.html

**Comment by ewbankkit** [comment]
@chgerkens Thanks for raising this issue.
AWS credentials are handled by the https://github.com/hashicorp/aws-sdk-go-base library.
I have opened an issue - https://github.com/hashicorp/aws-sdk-go-base/issues/87.

**Comment by gdavison** [comment]
This is also affected by aws/aws-sdk-go-v2#575

**Comment by gdavison** [comment]
Implemented in aws/aws-sdk-go-v2#1588

**Comment by wlami** [comment]
This has been implemented in #25368 and released in [v4.29.0 of the Terraform AWS Provider](https://github.com/hashicorp/terraform-provider-aws/blob/v4.29.0/CHANGELOG.md).
